pkg/cache/v2/cache.go

```diff
@@ -72,6 +87,21 @@ type Response interface {
 	GetVersion() (string, error)
 }
 
+// DeltaResponse is a wrapper around Envoy's DeltaDiscoveryResponse
+type DeltaResponse interface {
+	// Get the constructed DeltaDiscoveryResponse
+	GetDeltaDiscoveryResponse() (*discovery.DeltaDiscoveryResponse, error)
+
+	// Get te original Request for the Response.
+	GetDeltaRequest() *discovery.DeltaDiscoveryRequest
```

Comment:
Right, this should be treated as the most recent request. As far as my understanding of the protocol goes, the communication mechanism between the management server and the xDS client is still the same, but the management server is in charge of the delta's and spontaneously sending out the correct cache information based upon those deltas. And then of course the ability for clients to lazy load/request, etc...

But the reason this is here is because we still need the ability to track the the ACK/NACK i.e. the `response_nonce` and any error details that might come through. So this does become the most recent request on the stream

Reply:
Yeah I think that's fine, just make sure we capture that in the comments etc. The protocol difference is that SotW there is a sense that "this is request that requested this response", but for delta the "requested state" is the sum of all the requests on the stream. Because of that the term "original" seems more ambiguous because it could also refer to the first request of the stream

Reply:
Yup understood, I'll make sure in the comments that is noted.